content/recipes/typeorm-migration.md

```diff
@@ -0,0 +1,112 @@
+### TypeORM migration
+
+#### Objectives
+Generate and use migrations instead of syncing database. In dev and prod.
+
+#### Pre-requisites
+TypeORM installed: https://docs.nestjs.com/techniques/database
+
+#### Install
+
+**src/app.module.ts**
+```typescript
+import { DynamicModule, Module } from '@nestjs/common';
+import { AppController } from './app.controller';
+import { AppService } from './app.service';
+import { TypeOrmModule } from '@nestjs/typeorm';
+import * as ormconfig from './ormconfig';
+
+export function DatabaseOrmModule(): DynamicModule {
+  // we could load the configuration from dotEnv here,
+  // but typeORM cli would not be able to find the configuration file.
+
+  return TypeOrmModule.forRoot(ormconfig);
+}
+
+@Module({
+  imports: [
+    TypeOrmModule.forRoot(ormconfig)
+    // or
+    // DatabaseOrmModule(),
+  ],
+  controllers: [AppController],
+  providers: [AppService],
+})
+export class AppModule {}
+```
+
+
+**src/ormconfig.ts**
+```typescript
+import {ConnectionOptions} from 'typeorm';
+
+// You can load you .env file here synchronously using dotenv package (not installed here),
+// import * as dotenv from 'dotenv';
+// import * as fs from 'fs';
+// const environment = process.env.NODE_ENV || 'development';
+// const data: any = dotenv.parse(fs.readFileSync(`${environment}.env`));
+// You can also make a singleton service that load and expose the .env file content.
+// ...
+
+
+// Check typeORM documentation for more information.
+const config: ConnectionOptions = {
+  type: 'postgres',
+  host: 'localhost',
+  port: 5432,
+  username: 'postgres',
+  password: 'pwd',
+  database: 'migrationexample',
+  entities: [__dirname + '/**/*.entity{.ts,.js}'],
+
+  // We are using migrations, synchronize should be set to false.
+  synchronize: false,
+
+  // Run migrations automatically,
+  // you can disable this if you prefer running migration manually.
+  migrationsRun: true,
+  logging: true,
+  logger: 'file',
+
+  // Allow both start:prod and start:dev to use migrations
```

Comment:
As `start:dev` is now an alias for `nest start --watch` which runs `tsc --watch` under the hood, should this be updated to mention that only `dist + '/migrations/**/*.js'` need to be looked for, or should it be kept like this for the case of people still using `ts-node`?